.github/workflows/canary-integration-test.yml

```diff
@@ -1631,32 +1631,58 @@ jobs:
         with:
           use-tmate: ${{ secrets.USE_TMATE }}
 
+      - name: setup golang
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+
       - name: setup cluster resources
         uses: ./.github/workflows/canary-test-config
 
-      - name: allow holder pod deployment
-        run: sed -i "s|CSI_DISABLE_HOLDER_PODS|# CSI_DISABLE_HOLDER_PODS|g" "deploy/examples/operator.yaml"
-
       - name: set Ceph version in CephCluster manifest
         run: tests/scripts/github-action-helper.sh replace_ceph_image  "deploy/examples/cluster-test.yaml" "${{ github.event.inputs.ceph-image }}"
 
+      - name: install deps
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+        run: tests/scripts/github-action-helper.sh install_deps
+
+      - name: print k8s cluster status
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+        run: tests/scripts/github-action-helper.sh print_k8s_cluster_status
+
+      - name: build rook
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+        run: tests/scripts/github-action-helper.sh build_rook
+
+      - name: validate-yaml
+        run: tests/scripts/github-action-helper.sh validate_yaml
+
       - name: use local disk and create partitions for osds
         run: |
           tests/scripts/github-action-helper.sh use_local_disk
           tests/scripts/github-action-helper.sh create_partitions_for_osds
 
-      - name: deploy CSI hostNetworking disabled cluster
-        run: tests/scripts/github-action-helper.sh deploy_csi_hostnetwork_disabled_cluster
+      - name: deploy multus
+        run: tests/scripts/github-action-helper.sh deploy_multus
+
+      - name: deploy multus cluster
+        run: tests/scripts/github-action-helper.sh deploy_multus_cluster
 
       - name: wait for prepare pod
         run: tests/scripts/github-action-helper.sh wait_for_prepare_pod 2
 
       - name: wait for ceph to be ready
-        run: IS_POD_NETWORK=true tests/scripts/github-action-helper.sh wait_for_ceph_to_be_ready osd 2
+        run: IS_POD_NETWORK=true IS_MULTUS=true tests/scripts/github-action-helper.sh wait_for_ceph_to_be_ready osd 2
```

Comment:
Won't `IS_MULTUS=true` cause the test to fail because of the block below?

`tests/scripts/validate_cluster.sh`:
```
    if [ -n "$IS_MULTUS" ]; then
      echo "verifying csi holder interfaces (multus ones must be present)"
      kubectl -n rook-ceph exec -t ds/csi-rbdplugin-holder-my-cluster -- grep eth0 /proc/net/dev
      kubectl -n rook-ceph exec -t ds/csi-cephfsplugin-holder-my-cluster -- grep eth0 /proc/net/dev
      kubectl -n rook-ceph exec -t ds/csi-nfsplugin-holder-my-cluster -- grep eth0 /proc/net/dev
    fi
```

Reply:
yeah probably and also the routing part is still pending to add